### ISSUE TITLE
Code fails to start up with invalid tokenColorCustomizations

### ISSUE THREAD

**Issue by zachlitz** [Microsoft/vscode#43129]
- VSCode Version: 1.20.0 (commit c63189deaa)
- OS Version:  Win 10 Pro Insider Preview V1709

After the V1.20 update, VS Code will no longer do anything except render the top (native) toolbar/menu.

Developer console shows an error (I don't know if it is related). None of the menu commands work.

![image](https://user-images.githubusercontent.com/3002265/35940783-7ac7c266-0c0d-11e8-87d7-bbf4d4a6bedd.png)

Steps to Reproduce:

1. Open VS Code

Does this issue occur when all extensions are disabled?: Yes/No
Yes!.

Also, I uninstalled 32-bit version and installed 64-bit version.

**Comment by zachlitz** [comment]
I just tried Insiders build and that works (but 1.20.0 is still broken).

**Comment by ramya-rao-a** [comment]
What is the version of the Insiders that works?

**Comment by zachlitz** [comment]
The version of Insiders that works is 
* V1.20.0-insider 
* commit f9398
* date 2018-02-06 15:20:43Z
* Shell 1.7.9
* Renderer 58.0.3029.110
* Node 7.9.0

**Comment by ramya-rao-a** [comment]
Can you run `code   --user-data-dir <path to an empty folder>` and see if you still have the issue?

**Comment by zachlitz** [comment]
I deleted `users/me/.vscode` and that did not work.

But running `code --user-data-dir <path to an empty folder>` did work.

**Comment by ramya-rao-a** [comment]
@zachlitz That's good to know. That means something from the user settings is causing this. Can you share the contents of your settings file? You can find it at `%USERPROFILE%/AppData/Roaming/Code/User/settings.json` where `%USERPROFILE%` is usually `C:/Users/your-user-name`

**Comment by zachlitz** [comment]
```
// Place your settings in this file to overwrite the default settings
{
    "typescript.tsdk": "C:\\Users\\ZachLitz\\AppData\\Roaming\\npm\\node_modules\\typescript\\lib",
    "files.trimTrailingWhitespace": true,
    "window.zoomLevel": 0,
    "window.openFilesInNewWindow": "default",
    "window.openFoldersInNewWindow": "default",
    "explorer.autoReveal": false,
    "typescript.referencesCodeLens.enabled": true,
    "editor.wordWrap": "wordWrapColumn",
    "editor.renderWhitespace": "all",
    "editor.renderControlCharacters": false,
    "editor.renderIndentGuides": true,
    "window.title": "${dirty}${activeEditorMedium}${separator}${rootPath}",
    "vsicons.projectDetection.disableDetect": true,
    "[markdown]": {
        "editor.wordWrap": "on",
        "editor.wordWrapColumn":120,
        "editor.quickSuggestions": false
    },
    "editor.detectIndentation": false,
    "[typescript]": {

    },
    "jsDocCompletion.enabled": false,
    "workbench.colorCustomizations": {
        "statusBar.debuggingForeground": "#FFFFFF",
        "statusBar.debuggingBackground": "#FF00BF",
        "editorWarning.foreground": "#ffffff",
        "editor.findMatchHighlightBackground": "#686666",
        "editor.hoverHighlightBackground": "#505050",
        "editor.selectionHighlightBackground": "#03645c",
        "activityBar.background": "#ffffff",
        "activityBar.foreground": "#d40202",
        "editorBracketMatch.background": "#5c5b5b",
        "editorBracketMatch.border": "#00ff15",
        "editorLineNumber.foreground": "#2196f7"
    },
    "workbench.colorTheme": "Gloom",
    "files.associations": {
        "*.js.map": "json",
        "*.map": "json",
        "*.hhtml": "html",
        "*.cwhtml": "html"
    },
    "[json]": {

    },
    "workbench.iconTheme": "vscode-great-icons",
    "editor.fontSize": 13,
    "editor.minimap.enabled": false,
    "editor.cursorBlinking": "blink",
    "editor.cursorStyle": "line",
    "editor.renderLineHighlight": "all",
    "editor.showFoldingControls": "always",
    "git.enabled": false,
    "terminal.integrated.fontSize": 13,
    "vsicons.dontShowNewVersionMessage": true,
    "terminal.integrated.scrollback": 10000,
    "python.pythonPath": "C:\\Users\\ZachLitz\\AppData\\Local\\Programs\\Python\\Python36\\python.exe",
    "debug.hideActionBar": true,
    "tslint.alwaysShowRuleFailuresAsWarnings": true,
    "graphviz-preview.dotPath": "C:\\Program Files (x86)\\Graphviz2.38\\bin\\dot.exe",
    "extensions.ignoreRecommendations": false,
    "editor.tokenColorCustomizations": {
        "comments": "#9B9B9B",

        "textMateRules": [
            {

            }
        ]
    },
    "powermode.enabled": true,
    "powermode.presets": "particles",
    "powermode.enableShake": false,
    "winopacity.opacity": 10,
    "dotnet-test-explorer.build": false,
    "cSpell.userWords": [
        "Cityworks",
        "Nuget"
    ],
    "workbench.startupEditor": "welcomePage",
    "window.menuBarVisibility": "default",
    "editor.wordWrapColumn": 160
    // "terminal.external.windowsExec": "C:\\Windows\\sysnative\\bash.exe",
    // "terminal.integrated.shell.windows": "C:\\Program Files\\Git\\bin\\bash.exe"
}
```

**Comment by usernamehw** [comment]
`"editor.tokenColorCustomizations"` -> `"textMateRules"` is an empty object, which is probably the cause

**Comment by zachlitz** [comment]
That is it. Taking that out fixed the problem. Thanks!

**Comment by ramya-rao-a** [comment]
Thanks @usernamehw!
cc @aeschli for `editor.tokenColorCustomizations`

**Comment by aeschli** [comment]
Pushed a fix in master as well as release/1.20
